cli/src/main.rs

```diff
@@ -233,15 +237,40 @@ pub(crate) enum ItmSource {
 }
 
 fn main() -> Result<()> {
-    tracing_subscriber::fmt::fmt()
+    let project_dirs = directories::ProjectDirs::from("rs", "probe-rs", "probe-rs")
+        .expect("An application directory");
+    let directory = project_dirs.data_dir();
+    let logname = sanitize_filename::sanitize_with_options(
+        format!("{}.log", Local::now().timestamp_millis()),
+        sanitize_filename::Options {
+            replacement: "_",
+            ..Default::default()
+        },
+    );
+    std::fs::create_dir_all(&directory).expect("log directory could not be created.");
+
+    let log_path = directory.join(logname);
+    let log_file = File::create(dbg!(&log_path))?;
```

Comment:
```suggestion
    let log_file = File::create(&log_path)?;
```

Reply:
Would also be good to add the path to the error message here, e.g. with `.context()`.